### PR TITLE
Dev/capture eye data

### PR DESCRIPTION
**Pull request by joshanderson-kw**
Add EyeGazeData ROS message which contains eye gaze origin and direction, as well as info about what object was being looked at (if any).

Add EyeGazeData capture & publish C# script.

Eye calibration must be performed on the HL2 if it hasn't been already.
https://docs.microsoft.com/en-us/windows/mixed-reality/mrtk-unity/features/input/eye-tracking/eye-tracking-basic-setup?view=mrtkunity-2021-05#eye-calibration

### CODE REVIEW

File: unity/Hello_World/Assets/Scripts/EyeGazeCapture.cs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                                                                  eyeGazeProvider.HeadMovementDirection.y,
+                                                                  eyeGazeProvider.HeadMovementDirection.z);
+                Vector3Msg headVelocity = new Vector3Msg(eyeGazeProvider.HeadVelocity.x,
+                                                         eyeGazeProvider.HeadVelocity.y,
+                                                         eyeGazeProvider.HeadVelocity.z);
+                PointMsg hitObjectPosition = new PointMsg(eyeGazeProvider.HitInfo.point.x,
+                                                          eyeGazeProvider.HitInfo.point.y,
+                                                          eyeGazeProvider.HitInfo.point.z);
+
+                // Create the ROS std_msgs header with the EyeGazeProvider's timestamps
```
Comment (Purg) [inline unity/Hello_World/Assets/Scripts/EyeGazeCapture.cs:78]: We could add a TODO to use the common header gen util function you created in the other PR.

File: ros/angel_msgs/msg/EyeGazeData.msg [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+std_msgs/Header header
+
+# Eye gaze position
+geometry_msgs/Point gaze_origin
+
+# Eye gaze direction
+geometry_msgs/Vector3 gaze_direction
+
+# Head movement info
+geometry_msgs/Vector3 head_movement_direction
```
[the remark this answers is not in the document]
Comment (joshanderson-kw) [reply ros/angel_msgs/msg/EyeGazeData.msg:14]: Hmm I wasn't sure what info was necessary so I included most everything that was provided by the `IMixedRealityEyeGazeProvider`.

Can remove head movement direction and velocity if they are not useful.

[the remark this answers is not in the document]
Comment (Purg) [reply ros/angel_msgs/msg/EyeGazeData.msg:14]: It's likely fine to leave in there, I was just wondering if there was room for documenting what this was for. Sounds like we could comment that this is incidental information provided by the source sensor/API.

**Comment by Purg** [comment]
I think you will need to rebase on the updated master and resolve these conflicts such that the Unity scene is not in a bad state.

**Comment by joshanderson-kw** [comment]
> I think you will need to rebase on the updated master and resolve these conflicts such that the Unity scene is not in a bad state.

I think this should be good now. I built and tested on my end and things seem to be working.

**Review by Purg — APPROVED** [review]

**Pull request merged**